flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecLookupJoin.java

```diff
@@ -332,7 +319,7 @@ protected Transformation<RowData> createJoinTransformation(
         }
     }
 
-    private Transformation<RowData> createSyncLookupJoinWithState(
+    protected Transformation<RowData> createSyncLookupJoinWithState(
```

Comment:
Because only `StreamExecLookupJoin` needs stateful lookup join under certain circumstances. But `CommonExecLookupJoin#createJoinTransformation` is shared by both `BatchExecLookupJoin` and `StreamExecLookupJoin`. 

WDYT if we make the `createSyncLookupJoinWithState ` abstract and let subclasses override?

Reply:
I'm fine with that. Also don't forget to update `applyUpsertMaterialize` in `CommonExecSink`.